internal/service/managedobjectstorage/managed_object_storage.go

```diff
@@ -0,0 +1,360 @@
+					},
+					false,
+				),
+			},
+			"created_at": {
+				Description: "Creation time.",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+			"endpoint": {
+				Description: "Endpoints for accessing the Managed Object Storage service.",
+				Computed:    true,
+				Type:        schema.TypeSet,
+				Elem:        schemaEndpoint(),
+			},
+			"labels": utils.LabelsSchema("network gateway"),
+			"network": {
+				Description: "Attached networks from where object storage can be used. Private networks must reside in object storage region.",
+				Optional:    true,
+				Type:        schema.TypeSet,
+				Elem:        schemaNetwork(),
+			},
+			"operational_state": {
+				Description: "Operational state of the Managed Object Storage service.",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+			"region": {
+				Description: "Region in which the service will be hosted, see `upcloud_object_storage2_regions` data source.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			"updated_at": {
+				Description: "Creation time.",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+			"users": {
+				Description: "List of cloud Users allowed to use object storage. See `upcloud_managed_object_storage_user_access_key` for managing access keys.",
+				Optional:    true,
+				Type:        schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func schemaEndpoint() *schema.Resource {
+	return &schema.Resource{
+		Description: "OpenSearch index",
+		Schema: map[string]*schema.Schema{
+			"domain_name": {
+				Description: "Domain name of the endpoint.",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+			"type": {
+				Description: "Type of the endpoint (`private` / `public`).",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+		},
+	}
+}
+
+func schemaNetwork() *schema.Resource {
+	return &schema.Resource{
+		Description: "Managed Object Storage network",
+		Schema: map[string]*schema.Schema{
+			"family": {
+				Description: "Network family. IPv6 currently not supported.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						upcloud.IPAddressFamilyIPv4,
+						// IPv6 currently not supported
+					},
+					false,
+				),
+			},
+			"name": {
+				Description: "Network name. Must be unique within the service.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+					validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9_-]+$`), ""),
+				),
+			},
+			"type": {
+				Description: "Network type.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						"private",
+						"public",
+					},
+					false,
+				),
+			},
+			"uuid": {
```

Comment:
LBaaS uses `network` instead of `uuid` even though both have `uuid` in API. Otherwise the network sub-schema seems to be equal.

Reply:
Let's keep this as `uuid`.